/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto3";
import "SerializableSchema.proto";
package NES;

option java_package = "stream.nebula.protobuf";
option java_multiple_files = true;

message JavaUdfDescriptorMessage {
  message JavaUdfClassDefinition {
    string class_name = 1;
    bytes byte_code = 2;
  }

  string udf_class_name = 1;
  string udf_method_name = 2;
  bytes serialized_instance = 3;
  repeated JavaUdfClassDefinition classes = 4;
  SerializableSchema inputSchema = 5;
  SerializableSchema outputSchema = 6;
  string input_class_name = 7;
  string output_class_name = 8;
}

